android/src/main/kotlin/be/tramckrijte/workmanager/DebugHelper.kt

```diff
@@ -7,31 +7,56 @@ import android.os.Build
 import androidx.core.app.NotificationCompat
 import androidx.work.ListenableWorker
 import io.flutter.view.FlutterCallbackInformation
-import java.text.SimpleDateFormat
-import java.time.LocalDate
-import java.time.format.DateTimeFormatter
+import java.text.DateFormat
 import java.util.*
+import java.util.concurrent.TimeUnit.MILLISECONDS
+
+object ThumbnailGenerator {
+    fun mapResultToEmoji(result: ListenableWorker.Result): String =
+            when (result) {
+                is ListenableWorker.Result.Success -> "\uD83C\uDF89"
+                else -> "\uD83D\uDD25"
+            }
+
+    val workEmoji get() = listOf("\uD83D\uDC77\u200D♀️", "\uD83D\uDC77\u200D♂️").random()
+}
 
 object DebugHelper {
     private const val debugChannelId = "WorkmanagerDebugChannelId"
     private const val debugChannelName = "A helper channel to debug your background tasks."
-    private val debugDateFormatter = SimpleDateFormat("dd-MM-yyyy HH:mm:ss.ZZZZ")
+    private val debugDateFormatter = DateFormat.getDateTimeInstance(DateFormat.SHORT, DateFormat.MEDIUM)
 
     private val currentTime get() = debugDateFormatter.format(Date())
 
-    fun postTaskCompleteNotification(ctx: Context, dartTask: String, result: ListenableWorker.Result) {
-        postNotification(ctx, dartTask.hashCode(), currentTime, "Your work for $dartTask returned result: ${result.javaClass.simpleName}")
+    private fun mapMillisToSeconds(milliseconds: Long) = "${MILLISECONDS.toSeconds(milliseconds)} seconds."
+
+    fun postTaskCompleteNotification(ctx: Context, dartTask: String, fetchDuration: Long, result: ListenableWorker.Result) {
+        postNotification(
+                ctx,
+                dartTask.hashCode(),
```

Comment:
Maybe we could generate a unique id per 'run'. Because now 2 consecutive runs of the same task will override each other.  I would expect 1 notification per run, not 1 notification per defined task.

Reply:
Updated the PR ✅